tools/metricq-summary/src/main.cpp

```diff
@@ -0,0 +1,212 @@
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "run_command.hpp"
+#include "summary.hpp"
+
+#include <nitro/broken_options/parser.hpp>
+
+#include <metricq/logger/nitro.hpp>
+#include <metricq/simple.hpp>
+
+#include <iomanip>
+#include <iostream>
+#include <string>
+
+using Log = metricq::logger::nitro::Log;
+
+struct Config
+{
+    std::string url;
+    std::string token;
+    std::chrono::minutes timeout;
+    std::vector<std::string> metrics;
+    std::vector<std::string> cmdline;
+
+    Config(int argc, const char* argv[]);
+};
+
+Config::Config(int argc, const char* argv[])
+{
+    metricq::logger::nitro::set_severity(nitro::log::severity_level::info);
+
+    nitro::broken_options::parser parser;
+    parser.option("server", "The metricq management server to connect to.")
+        .default_value("amqp://localhost")
+        .short_name("s");
+    parser.option("token", "Token used for source authentication against the metricq manager.")
+        .default_value("summary");
+    parser.multi_option("metrics", "MetricQ metrics!").short_name("m");
+    parser.option("timeout", "Time-To-Live for data queue (in minutes)")
+        .default_value("30")
+        .short_name("T");
+    parser.toggle("verbose").short_name("v");
+    parser.toggle("trace").short_name("t");
+    parser.toggle("quiet").short_name("q");
+    parser.toggle("help").short_name("h");
+
+    try
+    {
+        auto options = parser.parse(argc, argv);
+
+        if (options.given("help"))
+        {
+            parser.usage();
+            std::exit(EXIT_SUCCESS);
+        }
+
+        if (options.given("trace"))
+        {
+            metricq::logger::nitro::set_severity(nitro::log::severity_level::trace);
+        }
+        else if (options.given("verbose"))
+        {
+            metricq::logger::nitro::set_severity(nitro::log::severity_level::debug);
+        }
+        else if (options.given("quiet"))
+        {
+            metricq::logger::nitro::set_severity(nitro::log::severity_level::warn);
+        }
+
+        metricq::logger::nitro::initialize();
+
+        url = options.get("server");
+        token = options.get("token");
+
+        try
+        {
+            for (size_t i = 0; i < options.count("metrics"); ++i)
+            {
+                metrics.push_back(options.get("metrics", i));
```

Comment:
Update Nitro and you can use [this](https://github.com/tud-zih-energy/nitro/blob/master/include/nitro/broken_options/options.hpp#L85)

Should reduce that to `metrics = options.get_all("metrics");`